pkg/reconciler/subscription/subscription.go

```diff
@@ -295,6 +210,38 @@ func (r *Reconciler) reconcile(ctx context.Context, subscription *v1alpha1.Subsc
 	} else if subStatus == corev1.ConditionFalse {
 		subscription.Status.MarkChannelFailed(subscriptionNotMarkedReadyByChannel, "Subscription marked by Channel as False")
 	}
+
+	return newReconciledNormal(subscription.Namespace, subscription.Name)
+}
+
+func (r *Reconciler) FinalizeKind(ctx context.Context, subscription *v1alpha1.Subscription) pkgreconciler.Event {
+	// Track the channel using the channelableTracker.
+	// We don't need the explicitly set a channelInformer, as this will dynamically generate one for us.
+	// This code needs to be called before checking the existence of the `channel`, in order to make sure the
+	// subscription controller will reconcile upon a `channel` change.
+	trackChannelable := r.channelableTracker.TrackInNamespace(subscription)
```

Comment:
Just curious why we set up the channel tracking in the FinalineKind in addition to the ReconcileKind? Seems curious and would be good to at least add a comment if it truly belongs here also.

Reply:
It was needed for the test only

Reply:
Also you would need it if you turned the controller back on and only had deleted subs

Reply:
Comment added.

Reply:
I'm still not quite understanding the need to track based on the channel. Purpose of the tracker is for the subscription to get requeued if the underlying Channel is changing, correct?

The way I read this code is that if everything goes well, as in we clean up after ourselves as necessary, in which case we should NOT be requeued anymore if that Channel changes (since we clean up, remove the finalizer and this object goes away)? If things don't go well, this object will get requeued anyways, because of the reconcile error?

For the cases where you have deleted subs upon restart, I would still expect those objects to get requeued just like any other object that still exists in the system?

Sorry, I'm just trying to understand what's up here :)

Reply:
The tracker sets up the ducktype informer for the gvr pointed at by the subscription so a little ways down when it goes to remove the sub from the channel, it needs the informer to make the patch